toc/rfc/rfc-draft-cc-blobstore-storage-cli.md

```diff
@@ -0,0 +1,68 @@
+# Meta
+[meta]: #meta
+- Name: Cloud Controller Blobstore Type: storage-cli
+- Start Date: 2025-07-18
+- Author(s): @johha, @stephanme
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: [community #1253](https://github.com/cloudfoundry/community/pull/1253)
+
+
+## Summary
+
+Add a new blobstore type `storage-cli` to the Cloud controller that is based on the Bosh storage CLIs. Long-term, the `storage-cli` blobstore type shall replace the blobstore type `fog`.
+The RFC also proposes to create a new "Storage CLI" area in the Foundational Infrastructure WG to allow cooperation of Bosh and CAPI teams and to consolidate the Bosh storage CLIs in one repository for easier code reuse.
+
+## Problem
+
+Cloud Controller uses the fog gem family to interface with the blobstores of different IaaS providers like Azure, AWS, GCP, and Alibaba Cloud.
+These Ruby gems are largely unmaintained, introducing risks such as:
+* Dependency on deprecated SDKs (e.g. Azure SDK for Ruby has reached EOL)
+* Blocking Ruby version upgrades
+* Potential for unpatched CVEs
+
+## Proposal
+
+Bosh faced similar issues, as it is also written in Ruby and interacts with blobstores. To address this, Bosh introduced standalone CLI tools which shell out from Ruby to handle all blobstore operations:
+- https://github.com/cloudfoundry/bosh-azure-storage-cli
+- https://github.com/cloudfoundry/bosh-s3cli
+- https://github.com/cloudfoundry/bosh-gcscli
+- https://github.com/cloudfoundry/bosh-ali-storage-cli
+
+These storage CLIs are implemented in Go and use the respective provider golang SDKs that are well supported for the foreseeable future.
+
+Cloud Controller shall implement a new blobstore type `storage-cli` that uses the mentioned storage CLIs for blobstore operations. Missing functionality needed by the Cloud Controller shall be added to the storage CLIs in a compatible way:
```

Comment:
CC supports sending arbitrary connection parameters to Fog:
1. https://github.com/cloudfoundry/capi-release/blob/9f69a02ef292c1e594f8da718ca524f378e573ae/jobs/cloud_controller_ng/spec#L553
2. https://docs.cloudfoundry.org/deploying/common/cc-blobstore-config.html#fog-aws-creds:~:text=(-,Optional)%20Provide%20additional%20configuration%20through%20the,hash%2C%20which%20is%20passed%20through%20to%20the%20Fog%20gem.,-Fog%20with%20AWS
3. https://github.com/cloudfoundry/cloud_controller_ng/blob/bb92b9ef3c0dc9808e9945c04848561aab9b9def/lib/cloud_controller/blobstore/fog/fog_client.rb#L202-L208

Operators could theoretically be configuring stuff in there that isn't supported by storage-cli (I haven't dug into the Fog library to see what all the connection options are).